expan/core/experiment.py

```diff
@@ -217,9 +219,24 @@ def sga(self, feature_name_to_bins):
                 subgroup = {'dimension': feature,
                             'segment': str(bin.representation)}
                 subgroup_data = bin.apply(self.data, feature)
+                if subgroup_data is None or len(subgroup_data) == 0:
+                    continue
                 subgroup_res = self._delta(method='fixed_horizon', data=subgroup_data,
                                            num_tests=len(self.report_kpi_names))
                 subgroup['result'] = subgroup_res
                 subgroups.append(subgroup)
 
+        if time_interval:
+            start = time_interval.get('start', '19000101')  # minimal date
+            end = time_interval.get('end', '22000101')  # maximal date
+            if 'date' in self.data:
+                subgroup_data = self.data[(start <= self.data.date) & (self.data.date <= end)]
+            else:
+                raise KeyError("No date column is provided in data for time-based SGA!")
+
+            if subgroup_data is not None:
+                subgroups.append({'dimension': 'date',
+                                  'segment': time_interval,
+                                  'result': self._delta(method='fixed_horizon', data=subgroup_data,
+                                                        num_tests=len(self.report_kpi_names))})
```

Comment:
oh and btw, I added a check: if subgroup data is None based on the filtering, we don't do ```_delta```(because it will return an error), but skip this subgroup and continue the next. Same applies for time interval based subgroup.

Reply:
@shansfolder I can move the subsetting data by time_interval before subgroups, so that we also can first filter on dates and then do subgroups. If only time_interval is set, there will be only 1 subgroup by date. Is that what you meant?